File: 1679. Max Number of K-Sum Pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        int count = 0; 
        int left = 0; 
        int right = nums.size() - 1;

        // Sort the array to use two-pointer technique
        sort(nums.begin(), nums.end());

        while (left < right) {
            int sum = nums[left] + nums[right];
            if (sum == k) {
                count++;
                left++;
                right--;
            } else if (sum > k) {
                right--;
            } else {
                left++;
            }
        }
        return count;
    }
};

File: 1679. Max Number of K-Sum Pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        count = 0
        left = 0
        right = len(nums)-1

        nums.sort()

        while left < right:
            sum = nums[left] + nums[right]

            if sum == k:
                count+=1
                left+=1
                right-=1
            elif sum > k:
                right-=1
            elif sum < k:
                left+=1
        
        return count
